superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx

```diff
@@ -40,16 +40,8 @@ const defaultProps = {
   query: {},
 };
 
-class ExploreResultsButton extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.getInvalidColumns = this.getInvalidColumns.bind(this);
-    this.renderInvalidColumnMessage =
-      this.renderInvalidColumnMessage.bind(this);
-  }
-
-  getColumns() {
-    const { props } = this;
+function ExploreResultsButton(props) {
```

Comment:
Please convert all of your functions to arrow functions (I'll mark the others with comments)

Reply:
All the others I completely agree with, but I think that this, being the functional component itself, does not need to be an arrow function. 

However, I think that you need to destructure props here, you can do that by console logging the props and writing them out individually, its more explicit.